### PR TITLE
[sgen] Add Ward annotations to sgen_get_total_allocated_bytes

### PR DESCRIPTION
**Pull request by lambdageek**
Attempt to fix https://jenkins.mono-project.com/job/test-mono-mainline-staticanalysis/

### CODE REVIEW

**Comment by lambdageek** [comment]
cc @akoeplinger

**Comment by naricc** [comment]
What does MONO_PERMIT actually do?

**Comment by lambdageek** [comment]
@naricc See https://github.com/evincarofautumn/Ward#enforcements

File: mono/sgen/sgen-gc.h [hunk 1]
```diff
@@ -1130,8 +1130,10 @@ typedef enum {
 } SgenAllocatorType;
 
 void sgen_clear_tlabs (void);
-void sgen_update_allocation_count (void);
-guint64 sgen_get_total_allocated_bytes (MonoBoolean precise);
+void sgen_update_allocation_count (void)
+	MONO_PERMIT (need (sten_world_stopped));
+guint64 sgen_get_total_allocated_bytes (MonoBoolean precise)
+	MONO_PERMIT (need (sgen_lock_gc, sgen_stop_world));
```
Comment (naricc) [inline mono/sgen/sgen-gc.h:1136]: So, I'm not sure if this matters, but sgen_get_total_allcoated_bytes only needs that lock if precise is true. I guess that is ok as a static over approximation though?

File: mono/sgen/sgen-gc.h [hunk 2]
```diff
@@ -1130,8 +1130,10 @@ typedef enum {
 } SgenAllocatorType;
 
 void sgen_clear_tlabs (void);
-void sgen_update_allocation_count (void);
-guint64 sgen_get_total_allocated_bytes (MonoBoolean precise);
+void sgen_update_allocation_count (void)
+	MONO_PERMIT (need (sten_world_stopped));
```
Comment (akoeplinger) [inline mono/sgen/sgen-gc.h:1134]: typo:
```suggestion
	MONO_PERMIT (need (sgen_world_stopped));
```

File: mono/sgen/sgen-gc.h (hunk 1 above)
Comment (lambdageek) [reply mono/sgen/sgen-gc.h:1136]: yea Ward is not at all flow sensitive, so that kind of conditional information is lost on it.

**Comment by lambdageek** [comment]
@monojenkins build failed

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**